### PR TITLE
initial bind of SourceHandle to BC for python

### LINKED ISSUE #9716: Expose SourceHandle to python for test script usage
**Is your feature request related to a problem? Please describe.**
I'm trying to fix test scripts, which provide script canvas files to script canvas editor components, but in doing so receive an error from python console. For example, when I try to get the script canvas file already referenced on a script canvas component using the EditorComponent class I receive the error "TypeError: get_component_property_value() missing 1 required positional argument: 'component_property_path'". This happens when providing a path as a string "Configuration|Source File"
 
**Describe the solution you'd like**
I would like to get/set the Source File property on a script canvas component using the same tools we use to set/get properties on other editor components.

**Describe alternatives you've considered**
N/A

**Additional context**
python test modules used: editor_python_test_tools.editor_entity_utils

### PR DESCRIPTION
**Pull request by carlitosan**
fixes #9716 

Signed-off-by: carlitosan <82187351+carlitosan@users.noreply.github.com>

### CODE REVIEW

File: AutomatedTesting/Gem/PythonTests/scripting/ScriptCanvas_TwoComponents_InteractSuccessfully.py [hunk 1]
```diff
@@ -55,15 +55,13 @@ def ScriptCanvas_TwoComponents_InteractSuccessfully():
     import azlmbr.math as math
     import azlmbr.asset as asset
     import azlmbr.bus as bus
+    import azlmbr.editor as editor
```
Comment (onecent1101) [inline AutomatedTesting/Gem/PythonTests/scripting/ScriptCanvas_TwoComponents_InteractSuccessfully.py:58]: Minor: unused import?

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Core.cpp [hunk 2]
```diff
@@ -287,6 +293,11 @@ namespace ScriptCanvas
 
     void SourceHandle::SanitizePaths()
     {
+        if (!m_relativePath.empty() && m_absolutePath.RelativePath() != m_relativePath)
+        {
+            m_absolutePath = m_relativePath;
```
Comment (onecent1101) [inline Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Core.cpp:298]: Question: do we still need abosolutePath? can we always use relative path instead?

Comment (carlitosan) [reply Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Core.cpp:298]: I'm investigating this.

File: AutomatedTesting/Gem/PythonTests/scripting/ScriptCanvas_TwoComponents_InteractSuccessfully.py (hunk 1 above)
Comment (carlitosan) [reply AutomatedTesting/Gem/PythonTests/scripting/ScriptCanvas_TwoComponents_InteractSuccessfully.py:58]: It's used here: editor.EditorComponentAPIBus(bus.Broadcast, 'BuildComponentPropertyList', component)

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/hydra_editor_utils.py [hunk 3]
```diff
@@ -307,6 +307,9 @@ def get_set_test(self, component_index: int, path: str, value: object, expected_
         if old_value is not None:
             print(f"SUCCESS: Retrieved property Value for {self.name}")
         else:
+            print("FAILURE: failed to find path in component. Existing Paths:\n")
+            paths = editor.EditorComponentAPIBus(bus.Broadcast, 'BuildComponentPropertyList', component)
+            print(paths)
```
Comment (cgalvan) [inline AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/hydra_editor_utils.py:312]: Nice, thanks for adding this!

File: Gems/ScriptCanvas/Code/Editor/Framework/Configuration.cpp [hunk 4]
```diff
@@ -187,14 +187,15 @@ namespace ScriptCanvasEditor
...
-                    ->DataElement(AZ::Edit::UIHandlers::Default, &Configuration::m_sourceHandle, "Source File", "Script Canvas source file associated with this component")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &Configuration::m_sourceHandle, "Source", "Script Canvas source file associated with this component")
                         ->Attribute("BrowseIcon", ":/stylesheet/img/UI20/browse-edit-select-files.svg")
                         ->Attribute("EditButton", "")
                         ->Attribute("EditDescription", "Open in Script Canvas Editor")
                         ->Attribute("EditCallback", &Configuration::OpenEditor)
                         ->Attribute(AZ::Edit::Attributes::AssetPickerTitle, "Script Canvas")
                         ->Attribute(AZ::Edit::Attributes::SourceAssetFilterPattern, "*.scriptcanvas")
                         ->Attribute(AZ::Edit::Attributes::ChangeNotify, &Configuration::OnEditorChangeSource)
+                        // ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::HideChildren) // or just ::Hide
```
Comment (cgalvan) [inline Gems/ScriptCanvas/Code/Editor/Framework/Configuration.cpp:198]: Can we remove this commented line?

**Review by cgalvan — APPROVED** [review]

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Core.cpp [hunk 5]
```diff
@@ -287,6 +293,11 @@ namespace ScriptCanvas
 
     void SourceHandle::SanitizePaths()
     {
+        if (!m_relativePath.empty() && m_absolutePath.RelativePath() != m_relativePath)
```
Comment (lumberyard-employee-dm) [inline Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Core.cpp:296]: The `RelativePath` calls within this function needs to be removed.

That is only returning the Relative portion of the absolute path.

**Review by lumberyard-employee-dm — CHANGES_REQUESTED** [review]

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Core.cpp (hunk 5 above)
Comment (carlitosan) [reply Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Core.cpp:296]: This part is about to get removed by a competing PR that will have to merge over this actually.

**Review by onecent1101 — APPROVED** [review]
Look good except that absolute path concern, seems like we are going to deal with it in a separate PR.

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Core.cpp (hunk 5 above)
Comment (carlitosan) [reply Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Core.cpp:296]: https://github.com/o3de/o3de/pull/10167  This one in particular, needs to go in first.

**Review by lumberyard-employee-dm — APPROVED** [review]

**Pull request merged**